### PR TITLE
code refine

### PR DESCRIPTION
**Pull request by yuzhou03**
- refine check_config,  use init_checkpoint instead of model_pt_path hardcode
-

### CODE REVIEW

File: training/benchmarks/cpm/pytorch/config/_base.py [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
 ## load and save args
 # Path to a directory containing a model checkpoint.
-init_checkpoint: str = None
+init_checkpoint: str = "cpm_model_states_medium.pt"
```
Comment (upvenly) [inline training/benchmarks/cpm/pytorch/config/_base.py:78]: 我有个想法，能不能把init_checkpoint的路径放在training/run_benchmarks/config/test_conf.py里面，目前cpm_model_states_medium.pt 放在base里面，base就不是base了，我理解base是模版，不该有这种特化的东西。

File: training/benchmarks/cpm/pytorch/run_pretraining.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def main():
     logger = cpm_driver.logger
     dist_pytorch.init_dist_training_env(config)
 
-    check.check_config(config, "cpm_model_states_medium.pt")
+    check.check_config(config, has_checkpoint=True)
```
Comment (upvenly) [inline training/benchmarks/cpm/pytorch/run_pretraining.py:47]: check_config 的逻辑不太对，ckpt的路径肯定存在于config里面，可以直接用来判断，has_checkpoint就算被赋值为True，不代表肯定有给ckpt的文件存在，可以直接用config里面的路径来做

File: training/benchmarks/driver/base.py [hunk 3]
```diff
@@ -21,6 +21,7 @@ def __init__(self, config, mutable_params):
         self.logger = None
 
     def setup_config(self, parser):
+
```
Comment (upvenly) [inline training/benchmarks/driver/base.py:24]: 这种不必要的空格就去掉，按照pylint格式来

File: training/benchmarks/driver/train_helper.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        config = self.config
+        worker_seeds, _ = distributed.setup_seeds(
+            config.seed, config.num_epochs_to_generate_seeds_for, config.device)
+        worker_seed = worker_seeds[0]
+
+        random.seed(worker_seed)
+        np.random.seed(worker_seed)
+
+        # TODO 需适配芯片
+        torch.manual_seed(worker_seed)
```
Comment (upvenly) [inline training/benchmarks/driver/train_helper.py:45]: 适配芯片的接口呢？init_rand_seed(seed, device) 这样？

File: training/benchmarks/driver/train_helper.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        torch.cuda.manual_seed(worker_seed)
+
+        torch.cuda.manual_seed_all(worker_seed)
+        torch.backends.cudnn.benchmark = False
+        torch.backends.cudnn.deterministic = True
```
Comment (upvenly) [inline training/benchmarks/driver/train_helper.py:50]: 这些只是NV上的用法，保留下其他接口的可拓展性？

File: training/benchmarks/glm/pytorch/config/_base.py [hunk 6]
```diff
@@ -10,7 +10,7 @@
 train_data: str = "ReCoRD/glm_train_eval_hdf5_sparse/train_hdf5/train_sparse.hdf5"
 eval_data: str = "ReCoRD/glm_train_eval_hdf5_sparse/eval_hdf5/eval_sparse.hdf5"
 output_dir: str = ""
-init_checkpoint: str = None
+init_checkpoint: str = "blocklm-large-blank/200000/mp_rank_00_model_states.pt"
```
Comment (upvenly) [inline training/benchmarks/glm/pytorch/config/_base.py:13]: 意见同cpm，glm别急，确定完再改。

File: training/benchmarks/glm/pytorch/run_pretraining.py [hunk 7]
```diff
@@ -40,8 +40,7 @@ def main():
 
     dist_pytorch.init_dist_training_env(config)
 
-    check.check_config(
-        config, "blocklm-large-blank/200000/mp_rank_00_model_states.pt")
+    check.check_config(config, has_checkpoint=True)
```
Comment (upvenly) [inline training/benchmarks/glm/pytorch/run_pretraining.py:43]: 意见同上。

**Comment by yuzhou03** [comment]
- [ ]  driver模块添加TrainHelper，logger, local_rank,seed，driver，得到训练统计信息get_finished_info
- [ ]  添加MutableParams类，区分必填参数和可选参数
- [ ]  fix 原有代码中的raise str的问题
- [ ]  优化check_config，去除model_pt_file入参。
- [ ]  run_pretraining.py中的checkpoint路径，由hardcode改为去取_base.py配置
- [ ]  部分函数添加参数type hint.

File: training/benchmarks/cpm/pytorch/config/_base.py (hunk 1 above)
Comment (yuzhou03) [reply training/benchmarks/cpm/pytorch/config/_base.py:78]: good idea. _base.py和 /test_conf.py 这两文件的边界就需要定义下。
_base.py中放超参数
test_conf.py 放其他case相关的参数
这是一种方案，test_conf.py中参数就会很多。添加一个case时，就不好确定哪些是必须要填写的。

File: training/benchmarks/cpm/pytorch/run_pretraining.py (hunk 2 above)
Comment (yuzhou03) [reply training/benchmarks/cpm/pytorch/run_pretraining.py:47]: 有的模型，想resnet50实现时，就没有下载checkpoint, 这时has_checkpoint赋值为False，表示不检查chkpt.

File: training/benchmarks/driver/base.py (hunk 3 above)
Comment (yuzhou03) [reply training/benchmarks/driver/base.py:24]: done
